storm-client/src/jvm/org/apache/storm/dependency/DependencyUploader.java

```diff
@@ -176,6 +193,22 @@ private boolean uploadDependencyToBlobStore(String key, File dependency)
         return uploadNew;
     }
 
+    private long doUploadDependencyJar(Path source, OutputStream destination) throws IOException{
```

Comment:
I think the idea here is great, but we should try to use existing tools instead. How about https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/IOUtils.html#copyLarge(java.io.InputStream,%20java.io.OutputStream,%20byte[])?

Reply:
clever! thanks for the advice...